torch/distributions/transforms.py

```diff
@@ -353,13 +361,13 @@ def __eq__(self, other):
         return isinstance(other, SigmoidTransform)
 
     def _call(self, x):
-        return torch.sigmoid(x)
+        return _clipped_sigmoid(x)
 
     def _inverse(self, y):
         return y.log() - (-y).log1p()
 
     def log_abs_det_jacobian(self, x, y):
-        return -(y.reciprocal() + (1 - y).reciprocal()).log()
+        return (y * (1 - y)).log()
```

Comment:
Yup, when we use `transforms`, it is better to compute log_det in term of `x` (to eliminate numerical precision while getting `y` from `x`). Thanks for providing a reference! I'll see if it really helps and gets back on this.

Reply:
Maybe we can discuss about this in #19802 first to let it merged first, then I'll make corresponding changes in this PR later.

Reply:
Sure. Now that I am looking at that PR again, I think we should probably just define a clamping function to clamp to `(0, 1)` and call that in both `_call` and `_inverse`, because users might be using the inverse transform with values that can be 0 or 1.

Reply:
Yeah, agree.